ministryofjustice/correspondence_tool_staff
# CT-2482 Add better tests for closing case; fix error when closing

## Description
I'd omitted to test the case closing properly; this PR addresses that, fixes the bug when closing a case, adds more validation to the closing states and checks to see the outcome is reflected on the case page
 
## Self-review checklist

 
### Screenshots

 
### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2482
 
### Deployment

 
### Manual testing instructions
Create a new offender SAR case, then move it through the states using the transition button until you get to close case. Check you can't submit an empty response, then fill in the date responded and continue. Again, check you can't submit an empty response and choose an outcome. Click to submit and ensure the values are shown on the case page and the case is closed successfully
